### PR TITLE
Don't use `callsubr` in global subroutines in CID-keyed CFF

### LINKED ISSUE #39: `callsubr` operator in global subroutines
Currently, compreffor may generate some global subroutines which contain `callsubr` operator.  But according to https://github.com/fonttools/fonttools/issues/856#issuecomment-283779329, it is invalid:

> I asked my colleagues and confirmed that it's invalid to use the `callsubr` operator in global subroutines.

In fact, I found the above fonttools' bug when trying to subset a font subroutinized by compreffor.

### PR DESCRIPTION
**Pull request by mashabow**
Fix #39

This patch is something straightforward rather than optimal, but I think it is enough.

### CODE REVIEW

**Comment by anthrotype** [comment]
Thank you! I wonder if you could also add a small test for this?

File: src/python/compreffor/pyCompressor.py [hunk 1]
```diff
@@ -786,7 +786,13 @@ def mark_reachable(cand_subr, fdidx):
                     # no room for this one
                     bad_substrings.append(subr)
 
-        bad_substrings.extend([s[1] for s in subrs]) # add any leftover subrs to bad_substrings
+        bad_substrings.extend([s[1] for s in subrs])  # add any leftover subrs to bad_substrings
+
+        if fdselect is not None:
+            # CID-keyed: Avoid `callsubr` usage in global subroutines
+            bad_lsubrs = Compreffor.collect_lsubrs_called_from(gsubrs)
+            bad_substrings.extend(list(bad_lsubrs))
```
Comment (anthrotype) [inline src/python/compreffor/pyCompressor.py]: You don't need to make a temporary list out of a set in order to use it in `extend` method, the latter should accept any iterable.

Comment (mashabow) [inline src/python/compreffor/pyCompressor.py]: Fixed, thank you!

**Comment by mashabow** [comment]
Yes, I'll add a test tomorrow.

**Comment by anthrotype** [comment]
If this is ready, I'd like to merge and tag a new compreffor release, since I am also about to tag a new fontmake release, and may as well bump its minimum requirements to incorporate this.

**Comment by mashabow** [comment]
It's ready to merge. Thanks!

**Pull request merged**